techstep-team/techstep_takehome
# Freedom Evenden Completed Challenge

### Running the application: 
- Make sure all dependencies are installed. backend: `npm install`, frontend: `yarn install`
- Navigate into backend and run `npm run dev` to run with nodemon
- In another terminal navigate to frontend and run `yarn start`

### testing 
- Navigate to site at http://localhost3000/
- Click the button to  redirect to Search page
- Try searching for users by name, Should display a list of users that match search string. Or it will display a message that no users matched the search
- You can click the techstep logo to navigate back to landing page.
- To view error handing go to `backend/routes/searcj.js` and change the query to cause an error
- Try searching again to see error message

### Major changes to Backend
- I changed the USERNAME env variable to DB_USER because process.env.USERNAME was reading the username of my computer. 
- Installed Nodemon for hot reload
- Added `dev` script to package.json

### Major changes to Frontend
- Installed Axios for handing API requests
- Installed React Router Dom for routing. 
- Made Button on Landing page redirect to `/search`
- Created a search page
